### DISCUSSION TITLE
[2장] useMemo 장단점과 useMemo 사용에 대해 어떤 의견을 가지고 있는지 말하시오.

### DISCUSSION THREAD

**Discussion by easyhyun00** [FE-tech-talk/TechTalk-React#14] [Q&A] [answered]
제목이 곧 내용 입니다

**Comment by easyhyun00** [comment]
렌더링을 최소한 줄이기 위해 useMemo 를 사용하여 최적화를 함

장점 - 최적화이기 때문에 불필요한 렌더링을 방지하여 성능에 도움이 되며, 비용도 props 에 대한 얕은 비교라서 비용이 그렇게 크지 않음
단점 - 메모리에 저장하는 것도 비용이 들고, 메모이제이션을 하는게 항상 성능에 좋은 것은 않아 불필요하며, 메모이제이션 로직을 추가하면 가독성이 떨어짐

**Comment by codefug** [comment, marked as answer]
안좋음
